### ISSUE TITLE
re-exports can't be declared `@public`

### ISSUE THREAD

**Issue by TkDodo** [webpro/knip#342]
knip shows false positives for an export that is annotated with `@public` if that export is a re-export.

This code works as expected:

```
// my-utils.ts

/** @public */
export const unusedExport = 5
```

This doesn't:

```
// my-utils.ts
export const unusedExport = 5

// my-barrel.ts

/** @public */
export { unusedExport } from './my-utils.ts'
```

## reproduction

I have a reproduction of the issue here:

https://github.com/TkDodo/knip-issue/tree/public-re-exports

> ⚠️ This is not on the `main` branch, but on the `public-re-exports` branch

### steps to reproduce

- check out the repo
- `pnpm install`
- `pnpm run knip`

### Observed behaviour

It errors with:

```
Unused exports (1)
unusedExport  packages/packageA/src/components/index.ts
```

### Expected behaviour

It should say:

```
✂️  Excellent, Knip found no issues.
```

### Further comments

If you follow the [instructions here](https://github.com/TkDodo/knip-issue/blob/public-re-exports/packages/packageA/src/components/index.ts#L4) and replace the re-export with a normal export declaration, you should see the error go away.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.40.2. See [Release 2.40.2](https://github.com/webpro/knip/releases/tag/2.40.2) for release notes._

**Comment by webpro** [comment]
Now they can :)

Thanks for the very helpful repro 🙏
